index.test.ts

```diff
@@ -8,188 +8,147 @@ import retry from 'async-retry';
-      // so we use tree-kill to kill the process and it's children
-      if (keystoneProcess.pid) {
-        await treeKill(keystoneProcess.pid);
-      }
-    };
-
-    await adminUIReady;
-  }
-
   if (mode === 'development') {
     // process.env.SESSION_SECRET is randomly generated for this
-
     test('start keystone in dev', async () => {
-      await startKeystone('dev');
-    });
-  }
-
-  if (mode === 'production') {
+      cleanupKeystoneProcess = await startKeystone('dev');
+    }, 40000);
+  } else if (mode === 'production') {
     const env = {
       NODE_ENV: 'production',
       SESSION_SECRET: randomBytes(32).toString('hex'),
     };
 
     test('build keystone', async () => {
-      const keystoneBuildProcess = promisifiedExecFile('yarn', ['build'], {
+      const keystoneBuildProcess = execFileAsync('yarn', ['build'], {
         cwd: projectDir,
         env: {
           ...process.env,
           ...env,
         },
       });
-      const logChunk = (chunk: any) => {
-        console.log(chunk.toString('utf8'));
-      };
-      keystoneBuildProcess.child.stdout!.on('data', logChunk);
-      keystoneBuildProcess.child.stderr!.on('data', logChunk);
+      keystoneBuildProcess.child.stdout!.pipe(process.stdout);
+      keystoneBuildProcess.child.stderr!.pipe(process.stdout);
       await keystoneBuildProcess;
-    });
+    }, 100000);
+
     test('start keystone in prod', async () => {
-      await startKeystone('start', {
+      cleanupKeystoneProcess = await startKeystone('start', {
         ...process.env,
         ...env,
       });
-    });
+    }, 30000);
   }
 
   describe.each(['chromium'] as const)('%s', (browserName) => {
     let page: playwright.Page = undefined as any;
     let browser: playwright.Browser = undefined as any;
+
     beforeAll(async () => {
-      await retry(async () => {
-        await deleteAllData(projectDir);
-        browser = await playwright[browserName].launch();
-        page = await browser.newPage();
-        page.setDefaultNavigationTimeout(6000);
-        await page.goto('http://localhost:3000');
-      });
+      await deleteAllData(projectDir);
+      browser = await playwright[browserName].launch();
+      page = await browser.newPage();
+      page.setDefaultNavigationTimeout(6000);
     });
-    test('init user', async () => {
-      await retry(async () => {
-        await page.fill('label:has-text("Name") >> .. >> input', 'Admin');
-        await page.fill(
-          'label:has-text("Email") >> .. >> input',
-          'admin@keystonejs.com'
-        );
-        await page.click('button:has-text("Set Password")');
-        await page.fill('[placeholder="New Password"]', 'password');
-        await page.fill('[placeholder="Confirm Password"]', 'password');
-        await page.click('button:has-text("Get started")');
-        await page.uncheck('input[type="checkbox"]', { force: true });
-        await Promise.all([
-          page.waitForNavigation(),
```

Comment:
`waitForNavigation` is unnecessary when `click` automatically waits for the selector